src/Hintman/Suggest/TrailingNewline.hs

```diff
@@ -0,0 +1,19 @@
+module Hintman.Suggest.TrailingNewline
+       ( suggestNoTrailingNewline
+       ) where
+
+import Hintman (LineNumber)
+
+import qualified Data.Text as T
+
+
+suggestNoTrailingNewline :: (FilePath, Text) -> Maybe (FilePath, LineNumber, Text)
```

Comment:
I think that type is not accurate here. Looks like you actually should return list of things because you can return multiple newlines.

But I guess that in general this type is not precise enough. Since suggestion API can work only with single line, it probably makes sense that at the beginning of processing file we will split file in lines and than run multiple suggestions on every line and only then we will aggregate results. So looks like that every suggestion function should be of type:

```haskell
suggestSomething :: Text -> Maybe Text
```

function from line to line. And then in one place we will call all such functions for every line.

Reply:
if `suggestNoTrailingNewline` becomes `Text -> Maybe Text` then I would be able to know if it's a newline in the middle of the file or not. Would that logic be delegated outside that function?

Reply:
so you mean
`(FilePath, [Text]) -> Maybe (FilePath, LineNumber, Text)` ?

what about
`(FilePath, [Text]) -> Maybe (FilePath, [LineNumber], Text)`?

There's also @willbasky suggestion to turn these into types, maybe `Input` and `Output`? although I'm not sure if they can be reused by other suggestion functions

Reply:
@sphaso Looks like the better type here is:

```haskell
suggestNoTrailingNewlines :: [(LineNumber, Text)] -> [Suggestion]
```

`FilePath` is meta information needed for API, so no need to have in this function. this part can be factored out. If no trailing newlines then the output will be just empty list.

Reply:
@ChShersh Currently `Suggestion` contains the `FilePath` (as per #16). I'll wait for #16 and #21 to reach some conclusion before refactoring